File: Backend/src/main/java/com/app/demo/service/AnimalService.java
package com.app.demo.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.app.demo.model.Animal;
import com.app.demo.repository.AnimalRepo;

@Service
public class AnimalService {

	@Autowired
	AnimalRepo repo;
	
	/** Create Animal logic
	 * @param animal
	 * @return
	 */
	public Animal addAnimal(Animal animal) {
		return repo.save(animal);
	}
	
	/** Read Animal logic
	 * @param animal
	 * @return
	 */
	public Animal getAnimal(String name) {
		return repo.findById(name).get();
	}
	
	/** Update Animal logic
	 * @param animal
	 * @return
	 */
	public Animal updateAnimal(Animal animal) {
		return repo.save(animal);
	}
	
	/** Delete Animal logic
	 * @param animal
	 */
	public void deleteAnimal(String name) {
		Animal animal = new Animal();
		animal.setName(name);
		repo.delete(animal);
	}
	
	public List<Animal> getAllAnimals() {
		return repo.findAll();
	}
}

File: Backend/src/main/resources/static/animalApp/src/app/animal.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Animal } from './animal';

@Injectable({
  providedIn: 'root'
})
export class AnimalService {

  private baseUrl: string = "http://localhost:9000/api/v1";
  private create: string = "create";
  private read: string = "read";
  private update: string = "update";
  private delete: string = "delete";
  private all: string = "all";

  constructor(private http: HttpClient) { }

  public addAnimal(animal: Animal): Observable<Animal> {
    return this.http.post<Animal>(`${this.baseUrl}/${this.create}`, animal);
  }

  public getAnimal(name: string): Observable<Animal> {
    return this.http.get<Animal>(`${this.baseUrl}/${this.read}/${name}`);
  }

  public updateAnimal(animal: Animal): Observable<Animal> {
    return this.http.put<Animal>(`${this.baseUrl}/${this.update}`, animal);
  }

  public deleteAnimal(name: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${this.delete}/${name}`);
  }

  public getAllAnimals(): Observable<Animal[]> {
    return this.http.get<Animal[]>(`${this.baseUrl}/${this.all}`);
  }
}
